pkg/controller/datavolume/external-population-controller.go

```diff
@@ -217,6 +311,10 @@ func (r *PopulatorReconciler) syncExternalPopulation(log logr.Logger, req reconc
 }
 
 func (r *PopulatorReconciler) updateStatusPhase(pvc *corev1.PersistentVolumeClaim, dataVolumeCopy *cdiv1.DataVolume, event *Event) error {
+	if isCDIVolumePopulator(dataVolumeCopy) {
```

Comment:
The `checkPopulationRequirements` function might be a bit rusty (we are still checking if the feature gate is enabled though it's been true by default since v1.24), but imo it makes more sense to include this check nested in the `else if supported` condition.